kornia/models/segmentor/segmentation_models.py

```diff
@@ -47,27 +51,30 @@ def __init__(
         )
 
     def preprocessing(self, input: Tensor) -> Tensor:
-        if self.preproc_params["input_space"] == "RGB":
+        # Ensure the color space transformation is ONNX-friendly
+        input_space = self.preproc_params["input_space"]
+        input = kornia.color.rgb_to_bgr(input) if input_space == "BGR" else input  # Assume input is already RGB if not BGR
+
+        # Normalize input range if needed
+        input_range = self.preproc_params["input_range"]
+        if input_range[1] == 255:
+            input = input * 255.0
+        elif input_range[1] == 1:
             pass
-        elif self.preproc_params["input_space"] == "BGR":
-            input = kornia.color.rgb_to_bgr(input)
         else:
-            raise ValueError(f"Unsupported input space: {self.preproc_params['input_space']}")
+            raise ValueError(f"Unsupported input range: {input_range}")
 
-        if self.preproc_params["input_range"] is not None:
-            if input.max() > 1 and self.preproc_params["input_range"][1] == 1:
-                input = input / 255.0
-
-        if self.preproc_params["mean"] is None:
-            mean = tensor(self.preproc_params["mean"]).to(input.device)
+        # Handle mean and std normalization
+        if self.preproc_params["mean"] is not None:
```

Comment:
`tensor([self.preproc_params["mean"]], device=input.device)`

In theory this allocates directly into the device